src/vs/platform/terminal/common/terminal.ts

```diff
@@ -3,9 +3,566 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import * as nls from 'vs/nls';
 import { Event } from 'vs/base/common/event';
-import { IProcessEnvironment } from 'vs/base/common/platform';
+import { IDisposable } from 'vs/base/common/lifecycle';
+import { RawContextKey } from 'vs/platform/contextkey/common/contextkey';
 import { URI } from 'vs/base/common/uri';
+import { IProcessEnvironment, OperatingSystem } from 'vs/base/common/platform';
+import { IEnvironmentVariableInfo } from 'vs/platform/terminal/common/environmentVariable';
+
+export const TERMINAL_VIEW_ID = 'terminal';
+
+/** A context key that is set when there is at least one opened integrated terminal. */
+export const KEYBINDING_CONTEXT_TERMINAL_IS_OPEN = new RawContextKey<boolean>('terminalIsOpen', false);
```

Comment:
All view and keybinding related constants need to live inside the workbench as they're dealing with workbench only concepts (not `TerminalProcess` concepts). Most of this file should move back and only the parts that are absolutely necessary should go into platform.

We also shouldn't be moving all of common/environmentVariable into platform as it contains things like `IEnvironmentVariableInfo` which defines things about the workbench widget.

Reply:
we need `IEnvironmentVariableInfo` within platform because it's of the `environmentVariableInfo` property on `ITerminalProcessManager`

Reply:
I don't think `ITerminalProcessManager` should be there either since that's the object that controls dealing with the off proc ptys, if it's needing in platform that seems like a bug/circular dep.